browser/ui/content_settings/brave_widevine_bundle_util.cc

```diff
@@ -0,0 +1,60 @@
+/* Copyright (c) 2019 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/ui/content_settings/brave_widevine_bundle_util.h"
+
+#include "base/path_service.h"
+#include "brave/common/pref_names.h"
+#include "chrome/browser/profiles/profile_manager.h"
+#include "chrome/common/chrome_paths.h"
+#include "components/prefs/pref_service.h"
+#include "content/public/browser/browser_thread.h"
+#include "content/public/browser/cdm_registry.h"
+#include "content/public/common/cdm_info.h"
+#include "media/base/decrypt_config.h"
+#include "media/base/video_codecs.h"
+#include "media/media_buildflags.h"
+#include "third_party/widevine/cdm/widevine_cdm_common.h"
+#include "widevine_cdm_version.h"  // NOLINT
+
+void MaybeRegisterWidevineCdm() {
+  DCHECK_CURRENTLY_ON(content::BrowserThread::UI);
+
+  PrefService* prefs = ProfileManager::GetActiveUserProfile()->GetPrefs();
+  // Only registers widevine cdm when user explicitly requested.
+  if (!prefs->GetBoolean(kWidevineOptedIn))
+    return;
+
+  base::FilePath cdm_path;
+  bool success = base::PathService::Get(chrome::FILE_WIDEVINE_CDM, &cdm_path);
+  DCHECK(success);
+  content::CdmCapability capability;
+  const base::Version cdm_version(WIDEVINE_CDM_VERSION_STRING);
+
+  // Add the supported codecs as if they came from the component manifest.
```

Comment:
this seems very fragile to me. Where did this list of codecs come from?

Reply:
Yes, I think so. I copied this logic from `IsWidevineAvailable()` in `chrome_content_client.cc`.

Reply:
Ah,, I think this can be improved by storing excluded cdminfo in `CdmRegistry` in at some place and can be reuse it here again.

Reply:
All this log is deleted and used cached info from `CdmRegistryImpl`.

Reply:
All duplicated code is deleted by caching CdmInfo that created by upstream.
https://github.com/brave/brave-core/pull/1475/commits/81ca1a17aeec7cfd7ceaf0cfbaf95fbae099fd4c